src/cmr/concepts/aclConcept.js

```diff
@@ -0,0 +1,344 @@
+      // the second element of the promise array
+      promises.push(
+        // eslint-disable-next-line no-promise-executor-return
+        new Promise((resolve) => resolve(null))
+      )
+    }
+
+    this.response = Promise.all(promises)
+  }
+
+  /**
+   * Log requested keys for metrics and debugging
+   * @param {Array} keys List of keys being requested
+   * @param {String} format Format of the request (json, umm, meta)
+   */
+  logKeyRequest(keys, format) {
+    // Prevent logging concept types, their meta keys are logged above
+    const filteredKeys = keys.filter((field) => CONCEPT_TYPES.indexOf(field) === -1)
+
+    filteredKeys.forEach((key) => {
+      console.log(`Request ${this.getRequestId()} from ${this.getClientId()} to [concept: ${this.getConceptType()}] requested [format: ${format}, key: ${key}]`)
+    })
+  }
+
+  /**
+   * Query the CMR JSON API endpoint to retrieve requested data
+   * @param {Object} searchParams Parameters provided by the query
+   * @param {Array} requestedKeys Keys requested by the query
+   * @param {Object} providedHeaders Headers requested by the query
+   */
+  fetchAcl(searchParams, requestedKeys, providedHeaders) {
+    this.logKeyRequest(requestedKeys, 'json')
+
+    // Construct the promise that will request data from the json endpoint
+    return aclQuery({
+      conceptType: this.getConceptType(),
+      params: pick(snakeCaseKeys(searchParams), this.getPermittedJsonSearchParams()),
+      nonIndexedKeys: this.getNonIndexedKeys(),
+      headers: providedHeaders
+    })
+  }
+
+  /**
+   * Return the response from the query to CMR
+   */
+  async getResponse() {
+    return this.response
+  }
+
+  /**
+   * Set the total number of records available for a given search to the json endpoint
+   * @param {Integer} itemCount The total number of records available for a given search
+   */
+  setJsonItemCount(itemCount) {
+    this.jsonItemCount = parseInt(itemCount, 10)
+  }
+
+  /**
+   * Parse and return the array of data from the nested response body
+   * @param {Object} jsonResponse HTTP response from the CMR endpoint
+   */
+  parseJsonBody(jsonResponse) {
+    const { data } = jsonResponse
+    const { items } = data
+
+    return items
+  }
+
+  /**
+   * Rename fields, add fields, modify fields, etc
+   * @param {Object} item The item returned from the CMR json endpoint
+   */
+  normalizeJsonItem(item) {
+    return item
+  }
+
+  /**
+   * Set the search after identifier for a search against against the json endpoint
+   * @param {Float} searchAfterIdentifier The search after identifier provided by CMR
+   */
+  setJsonSearchAfter(searchAfterIdentifier) {
+    this.jsonSearchAfterIdentifier = searchAfterIdentifier
+  }
+
+  /**
+   * Retrieve the request id header from the request
+   * @param {Object} headers The provided headers from the query
+   * @return {String} Request ID defined in the headers
+   */
+  getRequestId() {
+    const {
+      'CMR-Request-Id': requestId
```

Comment:
does case matter?

Reply:
This function is also re-use from Concept.js component, and this case it extracts exact format and rename